FlowCrypt/Functionality/Services/Compose Message Helper/ComposeMessageHelper.swift

```diff
@@ -268,16 +268,17 @@ final class ComposeMessageHelper {
                     fmt: shouldEncrypt ? .encryptInline : .plain
                 )
 
-                let threadId = self.messageIdentifier?.threadId?.stringId ?? threadId
+                let threadId = forceCreate ? nil : (self.messageIdentifier?.threadId?.stringId ?? threadId)
```

Comment:
I think `threadId` shouldn't be set to `nil` here, as in this case draft won't be added to current thread

Reply:
Well, forceCreate will be only called when `requested entity not found` error occurred.
When that thread/draft is deleted from Gmail side, currentThreadId should be invalid and need to create new one.

Reply:
I re-checked - if `threadId` is invalid then no error occurs and draft is saved to new thread.
app receives `requested entity not found` error only when draft was deleted, so `threadId` can be left as it is here

Reply:
Well, you can follow this way to check.
First create draft from iOS app and go back to draft screen
Open that draft again and delete this draft from browser(Gmail) while you are on compose screen.
Try to edit subject/text and api will now return `request entity not found error`
If we don't set threadId to nil in this case, this draft won't be saved and `request entity not found` error will occur permanentely.

Reply:
Ah, I see - I was testing with existing thread, but when creating new draft then this error occurs.
Let's leave it for now and see how it works, as it probably should be a quite rare case.

Reply:
ok